cosmwasm-plugin/main.go

```diff
@@ -0,0 +1,296 @@
+		return err
+	}
+
+	//err = modifyExistingFiles()
+	//if err != nil {return err}
+
+	return nil
+}
+
+func (p) ExecuteHookPre(hook plugin.ExecutedHook) error {
+	fmt.Printf("Executing hook pre %q\n", hook.Name)
+	return nil
+}
+
+func (p) ExecuteHookPost(hook plugin.ExecutedHook) error {
+	fmt.Printf("Executing hook post %q\n", hook.Name)
+	return nil
+}
+
+func (p) ExecuteHookCleanUp(hook plugin.ExecutedHook) error {
+	fmt.Printf("Executing hook cleanup %q\n", hook.Name)
+	return nil
+}
+
+// For future use ..
+func getChain(cmd plugin.ExecutedCommand, chainOption ...chain.Option) (*chain.Chain, error) {
+	var (
+		home, _ = cmd.Flags().GetString("home")
+		path, _ = cmd.Flags().GetString("path")
+	)
+	if home != "" {
+		chainOption = append(chainOption, chain.HomePath(home))
+	}
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+	return chain.New(absPath, chainOption...)
+}
+
+func installDependencies() error {
+	cmd := exec.Command("go", "get", "github.com/CosmWasm/wasmd/x/wasm@v0.41.0")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return fmt.Errorf("failed to install dependency: %w", err)
+	}
+	return nil
+}
+
+func replaceWordsInFile(filePath, targetWord, replacement string) error {
+	// Read content from the file
+	content, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return err
+	}
+
+	// Perform the replacement
+	modifiedContent := strings.ReplaceAll(string(content), targetWord, replacement)
+
+	// Write the modified content back to the file
+	err = ioutil.WriteFile(filePath, []byte(modifiedContent), 0644)
+	if err != nil {
+		return err
+	}
+
+	//fmt.Printf("Replaced planet with %s\n", replacement)
+
+	return nil
+}
+
+func createFile(inputFilename, outputDir, outputFilename string) error {
+
+	// Load the embedded template files
+	templatesFS, err := fs.Sub(templates, "wasm-wiring")
+	if err != nil {
+		fmt.Println("Error:", err)
+		return err
+	}
+
+	// Open the embedded input file
+	inputFile, err := templatesFS.Open(inputFilename)
+
+	if err != nil {
+		return err
+	}
+	defer inputFile.Close() // This line schedules the file to be closed when the function returns
+
+	// Read content from the embedded file
+	sourceContent, err := io.ReadAll(inputFile)
+	if err != nil {
+		return err
+	}
+
+	// Construct the output file path
+	outputPath := filepath.Join(outputDir, outputFilename)
+
+	// Write the embedded content to the output file
+	err = ioutil.WriteFile(outputPath, sourceContent, 0644)
+	if err != nil {
+		return err
+	}
+
+	// Replace chainName in the output file
+	err = replaceWordsInFile(outputPath, "planet", chainName)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Created %s in %s\n", outputFilename, outputDir)
+	return nil
+}
+
+func createNewFiles() error {
+	// Call createFile for each template file
+
+	// Create ante.go based on ante.txt content
+	err := createFile("ante.txt", "app", "ante.go")
+	if err != nil {
+		return err
+	}
```

Comment:
I think you could maybe make all `createFile` calls within a loop that iterates a struct with the data, something like:

```go
	files := []struct {
		template, outDir, outFile string
	}{
		{
			template:    "ante.txt",
			outDir:      "app",
			outFile:     "ante.go",
		},
		// TODO: Add the other templates here ...
	}
	for _, f := range files {
		err := createFile(f.template, f.outDir, f.outFile)
		if err != nil {
			return err
		}
	}
```

Reply:
Implemented changes with this [commit](https://github.com/ignite/plugins/commit/b8ea91ecc6f1a2be64689a052db72715f2ee8c59)